Sources/SwiftProtobufPluginLibrary/Descriptor.swift

```diff
@@ -436,14 +503,71 @@ public final class Descriptor {
   /// leading subset of `oneofs` (or the same if there are no synthetic entries).
   public private(set) lazy var realOneofs: [OneofDescriptor] = {
     // Lazy because `isSynthetic` can't be called until after `bind()`.
-    return self.oneofs.filter { !$0.isSynthetic }
+    return self.oneofs.filter { !$0._isSynthetic }
   }()
   /// The extension field defintions under this message.
   public let extensions: [FieldDescriptor]
 
   /// The extension ranges declared for this message. They are returned in
   /// the order they are defined in the .proto file.
-  public let extensionRanges: [ExtensionRange]
+  public let protoExtensionRanges: [ExtensionRange]
```

Comment:
This doesn't look right.  This is prefix "proto", but it isn't the one return proto message objects.  Is there something we can do for better naming going forward?

Can you overload properties on return type and @_disfavoredOverload?

Reply:
> Can you overload properties on return type and @_disfavoredOverload?

Sadly this is not possible.

I don't really like the name either but I'm not sure what a better alternative is, so happy to receive suggestions :D 
A maybe obvious one is `descriptorExtensionRanges` - not nice to repeat `descriptor` (because uses will likely be `descriptor.descriptorExtensionRanges`) but not the end of the world and may be clearer.

Reply:
Ugh.  maybe `extensionRangeDescriptors`?  Both options are non idea.

Reply:
What about `messageExtensionRanges` (following the wording used in the documentation for the prop)?

Reply:
That works and seems better then the other two.